web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         this.navigationService.getSubmissionId$(),
       ]).subscribe(() => this.cancelReposition())
     );
+
+    this.subscription.add(
+      this.submissionService.getSelectedSubmission$().subscribe(submission => {
+        if (submission instanceof Submission) {
+          this.submission = submission;
+          if (this.showSubmissionGeometry === true) {
+            this.removeSubmissionResultsOnMap();
+            this.addSubmissionResultsOnMap();
+          }
+        }
+      })
+    );
+
+    this.subscription.add(
+      this.navigationService.getSubmissionId$().subscribe(id => {
```

Comment:
We already subscribe to changes to submissionId above, perhaps this should live there?